src/patchkernel/patch_kernel_parallel.cpp

```diff
@@ -3088,6 +3465,401 @@ void PatchKernel::clearGhostCellOwners()
+}
+
+/*!
+	Update the information needed for ghost vertex data exchange.
+*/
+void PatchKernel::updateGhostVertexExchangeInfo()
+{
+	// Patch information
+	int patchRank = getRank();
+
+	// Idenfity owners of target and source vertices
+	std::unordered_map<long, int> exchangeVertexOwners = evaluateExchangeVertexOwners();
+
+	//
+	// Clear ghosts
+	//
+
+	// List of vertices that are no more ghosts.
+	//
+	// Previous ghost vertices will be converted to internal vertices.
+	std::vector<long> previousGhosts;
+	previousGhosts.reserve(getGhostVertexCount());
+	for (VertexConstIterator vertexItr = ghostVertexBegin(); vertexItr != ghostVertexEnd(); ++vertexItr) {
+		long vertexId = vertexItr.getId();
+		if (exchangeVertexOwners.count(vertexId) != 0) {
+			continue;
+		}
+
+		previousGhosts.push_back(vertexId);
+	}
+
+	// Convert previous ghosts to internal vertices
+	for (long vertexId : previousGhosts) {
+		ghostVertex2InternalVertex(vertexId);
+	}
+
+	//
+	// Initialize ghost
+	//
+
+	// List of new ghost vertices
+	//
+	// Vertices need to be sorted with their order in the storage.
+	std::map<std::size_t, long> newGhosts;
+	for (const auto &entry : exchangeVertexOwners) {
+		int vertexOwner = entry.second;
+		if (vertexOwner == patchRank) {
+			continue;
+		}
+
+		long vertexId = entry.first;
+		VertexIterator vertexItr = m_vertices.find(vertexId);
+		if (!vertexItr->isInterior()) {
+			continue;
+		}
+
+		newGhosts.insert({vertexItr.getRawIndex(), vertexId});
+	}
+
+	// Set the owner of the existing ghosts
+	for (VertexConstIterator vertexItr = ghostVertexBegin(); vertexItr != ghostVertexEnd(); ++vertexItr) {
+		long vertexId = vertexItr->getId();
+		int vertexOwner = exchangeVertexOwners.at(vertexId);
+		setGhostVertexOwner(vertexId, vertexOwner);
+	}
+
+	// Create new ghosts
+	//
+	// The list of ghosts is processed backwards to reduce the number of
+	// vertices that need to be swapped (there is a high probability that
+	// most of the ghost vertices are already at the end of the storage).
+	//
+	// If a vertex is already a ghost, we still need to update its owner.
+	for (auto iter = newGhosts.rbegin(); iter != newGhosts.rend(); ++iter) {
+		long vertexId = iter->second;
+		int vertexOwner = exchangeVertexOwners.at(vertexId);
+		internalVertex2GhostVertex(vertexId, vertexOwner);
+	}
+
+	//
+	// Identify exchange targets
+	//
+
+	// Clear targets
+	m_ghostVertexExchangeTargets.clear();
+
+	// Update targets
+	for (const auto &entry : m_ghostVertexOwners) {
+		int ghostVertexRank = entry.second;
+		long ghostVertexId = entry.first;
+		m_ghostVertexExchangeTargets[ghostVertexRank].push_back(ghostVertexId);
+	}
+
+	// Sort the targets
+	for (auto &entry : m_ghostVertexExchangeTargets) {
+		std::vector<long> &rankTargets = entry.second;
+		std::sort(rankTargets.begin(), rankTargets.end(), VertexPositionLess(*this));
+	}
+
+	//
+	// Identify exchange sources
+	//
+	// Since vertices are shared among elements, data exchange does not involve
```

Comment:
This comment is really clear, maybe can be included in some method documentation?

For instance in a getGhostTarget/SourceVertexExchangeInfo()? Or in the documentation of this method?

Does it make sense?